lib/matplotlib/widgets.py

```diff
@@ -3026,26 +3039,11 @@ def _release(self, event):
         xy1 = self.ax.transData.transform([x1, y1])
         self._eventrelease.x, self._eventrelease.y = xy1
 
-        # calculate dimensions of box or line
-        if self.spancoords == 'data':
-            spanx = abs(self._eventpress.xdata - self._eventrelease.xdata)
-            spany = abs(self._eventpress.ydata - self._eventrelease.ydata)
-        elif self.spancoords == 'pixels':
-            spanx = abs(self._eventpress.x - self._eventrelease.x)
-            spany = abs(self._eventpress.y - self._eventrelease.y)
-        else:
-            _api.check_in_list(['data', 'pixels'],
-                               spancoords=self.spancoords)
-        # check if drawn distance (if it exists) is not too small in
-        # either x or y-direction
-        minspanxy = (spanx <= self.minspanx or spany <= self.minspany)
-        if (self._drawtype != 'none' and minspanxy):
-            for artist in self.artists:
-                artist.set_visible(False)
+        if self._drawtype != 'none' and self._selector_too_small():
             if self._selection_completed:
                 # Call onselect, only when the selection is already existing
                 self.onselect(self._eventpress, self._eventrelease)
-            self._selection_completed = False
+            self.clear()
```

Comment:
Is this change intended here? It's not a refactoring but changes code, e.g. `clear()` calls `update()` and then we call update again below in 3501.

Reply:
I don't remember what is the purpose of the `visible` attribute but another consequence of using `clear` is that it changes the value of the `visible` attribute from `True` to `False` and I am not sure if this desirable - this would need to be checked.

Reply:
For the record: it seems that the `visible` attribute is only used by the selector itself (and should be private) and it is reset in `_press`, so the point I made here is not an issue!